spec/controllers/annotations_controller_spec.rb

```diff
@@ -126,40 +58,49 @@
       "@type" => "oa:Tag",
       "chars" => "scene1"
     }
-
   end
 
   shared_examples 'invalid annotation auth' do
-    it 'Authentication issues with endpoint for JWT' do 
-      scenarios = []
-      scenarios.push({ group_id: "11", has_access: "true", endpoint: /testserver.com/})
-      scenarios.push({ group_id: "10", has_access: "false", endpoint: /testserver.com/})
-      scenarios.push({ group_id: nil, has_access: nil, endpoint: /testserver.com/})
-
-      scenarios.each do |scenario|
-        group_id = scenario[:group_id]
-        has_access = scenario[:has_access]
-        endpoint = scenario[:endpoint]
+    before(:all) do
+      @scenarios = [
+        { group_id: "11", has_access: "true", endpoint: /testserver.com/ },
+        { group_id: "10", has_access: "false", endpoint: /testserver.com/ },
+        { group_id: nil, has_access: nil, endpoint: /testserver.com/}
+      ]
+    end
+
+    it 'returns 403 when jwt auth is enabled' do
+      Rails.application.config.use_jwt_auth = true
+
+      @scenarios.each do |scenario|
+        group_id, has_access, endpoint = scenario.values_at(:group_id, :has_access, :endpoint)
         stub_anno_auth(group_id, has_access, endpoint)
         request_proc.call
         expect(response.status).to eq(403)
       end
     end
+
+    it 'returns normal code when jwt auth is disabled' do
+      Rails.application.config.use_jwt_auth = false
+
+      @scenarios.each do |scenario|
+        group_id, has_access, endpoint = scenario.values_at(:group_id, :has_access, :endpoint)
+        stub_anno_auth(group_id, has_access, endpoint)
+        request_proc.call
+        expect(response.status).to eq(normal_status)
+      end
+    end
```

Comment:
The two tests seem to be doing the same thing. Can we combine into one function and pass a parameter?

Reply:
They have some common steps but certainly are testing different things (or same thing in different contexts), so we'd better not combine it to one example. Alternatively, we could create a function, for example, `iterate_over_scenarios_with_stubs()`, but I fear it will actually reduce readability, considering we are already using "shared examples" and "let" constructs which are not super-intuitive. Do you have a suggestion in mind?

Reply:
I was thinking the second suggestion at first, but if readability is an issue, then lets leave it as is.